osmosis-xml/src/main/java/org/openstreetmap/osmosis/xml/v0_6/impl/BaseXMLReader.java

```diff
@@ -71,7 +111,7 @@ protected void parseXML(final InputStream stream, final DefaultHandler handler)
      * @param handler A {@link DefaultHandler}
      */
     protected void handleXML(final DefaultHandler handler) {
-        try (InputStream stream = this.getInputStream()) {
+        try (InputStream stream = this.inputStream) {
             try (InputStream compressionStream = new CompressionActivator(this.method)
```

Comment:
We had try with resources before, meaning that the inputstream would be eventually closed, etc.
I don't see anything which would close the FileInputStream you've created earlier on: ` this.inputStream = new FileInputStream(file);`

Reply:
I don't see how it was closed in current code. I just added a new constructor, moved getInputStream() to the default constructor, and edited some print information. 

If it can be closed in current code, it will works fine in my new code.